docs/how-to/hip_runtime_api/call_stack.rst

```diff
@@ -0,0 +1,132 @@
+.. meta::
+    :description: This page lists graph safe ROCm libraries.
+    :keywords: AMD, ROCm, HIP, hipGRAPH
+
+********************************************************************************
+Call stack
+********************************************************************************
+
+The call stack and the program counter work together to manage the flow of
+control in a program, especially in the context of function calls, loops, and
+branching.
+
+The call stack is a concept in computer science, representing the stack data
+structure that stores information about the active subroutines or functions.
+Each time a function is called, a new frame is added to the top of the stack,
+containing information such as local variables, return addresses and function
+parameters. When the function execution completes, the frame is removed from the
+stack. This concept allows the program to return to the calling function and
+continue executing from where it left off.
+
+The program counter keeps track of the current or next instruction to
+be executed, while the call stack maintains the history of function calls,
+return addresses, and local variables. Together, they allow a program to execute
+complex control flow structures, handle function calls and return to the
+appropriate location in the code after a function completes.
+
+The call stack in GPU programming has extra complexity due to parallel execution
+also NVIDIA and AMD GPUs are using different approaches. NVIDIA GPUs have the
+independent thread scheduling feature since Volta GPU, where every thread has
+its own call stack and effective program counter. At AMD GPUs every
+warp/wavefront has its own call stack and program counter.
```

Comment:
Updated.
I would highlight the difference, perhaps some users can have a better intuition about it.

Reply:
You are right, the difference is worth highlighting, however currently it reads as if this would somehow affect the size of the call stack for the different implementations ("one is per thread, the other per wavefront"), which is not entirely true. Both implementations still need to store the variables etc per thread, it's just that on NVIDIA every thread has an "individual" call stack, whereas on AMD they are grouped per wavefront.

Reply:
I rephased a bit. Can you check it, please?